R/mainTests.r

```diff
@@ -898,6 +899,120 @@ region_col_completed <- function(data) {
+  if (all(c("old_la_code", "la_name", "new_la_code") %in% names(data))) {
+
+    # Check to see all columns are completed for rows where at least one is completed
+
+    la_all_complete_gathering <- function(data) {
+      if (is.na(data[["old_la_code"]])) {
+        old_completed <- FALSE
+      } else {
+        old_completed <- TRUE
+      }
+
+      if (is.na(data[["la_name"]])) {
+        name_completed <- FALSE
+      } else {
+        name_completed <- TRUE
+      }
+
+      if (is.na(data[["new_la_code"]])) {
+        new_completed <- FALSE
+      } else {
+        new_completed <- TRUE
+      }
+
+      la_completed_row <- list(old_completed, name_completed, new_completed)
+
+      return(la_completed_row)
+    }
+
+    # collate matrix of whether every col is completed or not per row
+
+    pre_pre_result <- t(matrix(unlist(apply(data, 1, la_all_complete_gathering)), nrow = 3))
+
+    # check using the matrix if all data rows are either fully completed or fully blank
+
+    la_all_complete_check <- function(results_row) {
+      if (all(results_row == TRUE) | all(results_row == FALSE)) {
+        return(TRUE)
+      } else {
+        return(FALSE)
+      }
+    }
+
+    pre_result <- apply(pre_pre_result, 1, la_all_complete_check)
+
+    if (all(pre_result == TRUE)) {
+      output <- list(
+        "message" = "Where one of the local authority columns is completed, the other two columns are also completed.",
+        "result" = "PASS"
+      )
+    } else {
+      output <- list(
+        "message" = "Where any of the three local authority columns have values for a row, all columns should be filled in.", # not sure how to word best
+        "result" = "FAIL"
+      )
+    }
+  } else {
+    if (any(c("old_la_code", "la_name", "new_la_code") %in% names(data))) {
+
+      # list of permutations for columns missing where at least one is present
+
+      if (!("la_name" %in% names(data)) && !("new_la_code" %in% names(data))) {
+        output <- list(
+          "message" = "Where old_la_code is included in the data file, la_name and new_la_code should also be included.",
+          "result" = "FAIL"
+        )
+      } else {
+        if (!("la_name" %in% names(data)) && !("old_la_code" %in% names(data))) {
+          output <- list(
+            "message" = "Where new_la_code is included in the data file, la_name and old_la_code should also be included.",
+            "result" = "FAIL"
+          )
+        } else {
+          if (!("new_la_code" %in% names(data)) && !("old_la_code" %in% names(data))) {
+            output <- list(
+              "message" = "Where la_name is included in the data file, new_la_code and old_la_code should also be included.",
+              "result" = "FAIL"
+            )
+          } else {
+            if (!("new_la_code" %in% names(data))) {
+              output <- list(
+                "message" = "Where la_name and old_la_code are included in the data file, new_la_code should also be included.",
```

Comment:
Our guidance page advises to leave cells blank if using legacy LAs without a 9-digit code - will we need to change to suggest they use ":" instead? (or now that we have the LA lookup file, could remove that bit of guidance entirely?)

Reply:
Ah yes of course!

In the latest commit I've stripped out the 'completed' part of this test, so now it's just a check for all columns being there if any are. I've done the same for the  region_completed test too.

Reply:
Sorry - to be clear, the guidance is still good to stay as it is, I've just changed this check and the region_completed check to match it.

I've also done a fix to the la_combinations test that the new test data flagged (it would fall over if some, but not all, la columns were present).